docs/5_2_scheduled_execution_api.adoc

```diff
@@ -68,22 +68,26 @@ This function can be called before scheduling a model partition for variables as
 Function <<get-and-set-variable-values,`fmi3Get{VariableTypeExclClock}`>>::
 These functions can be called after the computation of a model partition for variables assigned to that model partition via its associated <<Clock>> and all variables not associated to a <<Clock>> (global variables).
 
-+
-Set/get operations must be atomic for a single variable.
```

Comment:
I learned in the discussion that "atomic" is the right term for the intended behaviour. When we delete it here, shouldn't we used it somewhere else in the text?

Reply:
@IZacharias @andreas-junghanns  ... what do you think? I wouldn't miss it

Reply:
@IZacharias & @MBlesken : This is the last pending issue for this PR and I think this is very important to define clearly. I have made a proposal in this PR now to require the importer to ensure get/set calls are not preempted. Please fix this if this is wrong from your point of view. If you agree, we can mark this conversation as resolved and merge.

Reply:
@andreas-junghanns Sorry, I needed to undo your commit "Require importer to guarantee atomicity of get/set operations". It's not the importers' responsibility. I cannot fix it just by replacing "importer" to "FMU". I am working on a new commit to guide the reader a bit more ...

Reply:
I am afraid that people might draw the wrong conclusions from 
> Set/get operations must be atomic for a single variable

It could give the impression that lockPreemption calls are always necessary. But in fact lockPreemption should be used as rarely as possible. I would rather insert some recommendations and give additional hints. 

In fact, consistency problems can only occur for "shared" variables, i.e. for variables that are accessed from several tasks. These cases should be considered:

1.  A variable is associated to more than one clock (not recommended!). Then the importer might call the set/get operation for the same variable in more than one task. This also means that the execution of a model partition might be interrupted by a set/get for a variable that is used in this model partition.
2. A variable is associated to only one clock (one model partition), but the FMU uses the variable internally in more than one model partition. In this case, the importer calls the set operation only in one task (the task in which it also calls fmiActivateModelPartition() for the associated clock).
3. A variable is associated to only one clock (one model partition), and the FMU uses the variable in this one model partition only.

In these cases, it is up to the FMU to decide how to handle this.

- In (1.) atomicity is necessary if the FMU wants to ensure consistency.
- In (2.), if the variable is associated to the clock with the highest priority, get and set are never interrupted, at least not by a task that also accesses this variable.
But if the FMU requires consistency, the FMU's code snippets that "transfer" data from one model partition to another model partition must be "smart". Note that this might also affect “internal” variables with no entry in the modelDescription.xml. Consistency can be assured, for example, by using lockPreemption/unlockPreemption (see example in 5.1.1 Preemption Support), but other ways are also possible. 
- In (2.), if the variable is associated to the clock with a lower priority, get and set can be interrupted. Only the FMU knows if an interruption might lead to a consistency problem or not. Here again it is up to the FMU to secure its code.
- In (3.) it is no problem at all if set/get are interrupted by some other task.

Reply:
> @andreas-junghanns Sorry, I needed to undo your commit "Require importer to guarantee atomicity of get/set operations". It's not the importers' responsibility. I cannot fix it just by replacing "importer" to "FMU". I am working on a new commit to guide the reader a bit more ...

Can you explain why the FMU should be responsible?
If this is required, then the following will happen:
 - importer calls fmi3GetXXX
 - FMU calls lock_preemption
 - FMU computes return values (in SE, this is not heavy, the activation call is compute intensive)
 - FMU calls unlock_preemption
 - FMU returns

If we require the importer to ensure atomicity, the fmi3GetXXX call is simpler and more efficient because the preemption mechanisms is not required. Why can´t the importer handle the preemption functionality right away?

Reply:
@IZacharias : our posts crossed paths, sorry.
Refrasing your explanation: "Asking the FMU to lock/unlock preemption is more efficient because only in rare cases such locking is required and only the FMU knows these internals. The importer would have to be more cautious and lock more often to be sure."

But: why does an FMU need to lock during get/set operations at all? The importer is in control, no model partition can compute at this moment, right? (The issue of race conditions is not part of this discussion, FMUs must handle these for sure, otherwise preemption can never work.)

Reply:
@andreas-junghanns the clue is that lock preemptions in general should be avoided and only the FMU exporter can know where locks are actually needed. @IZacharias just tried to explain this. Is this helping? Imagine the importer had to guess where there could be potential violations. This would end up in locks for whole task bodies of all tasks except the highest priority task. That would clearly be absurd with respect to the intended preemptability of tasks.

Now, @IZacharias explanations hopefully help to understand how an FMU has to be implemented with respect to preemptability. (I also tried to improve the understanding with this PR.) My question is: Where does the implementers guide pick up on this? And how much explanation and examples need to be in the specification? (Are there specific parts in Irina's explanations that you missed and should be included in the specification?) FYI @pmai